lib/AST/Decl.cpp

```diff
@@ -8253,6 +8258,13 @@ ParamDecl *ParamDecl::createParsed(ASTContext &Context, SourceLoc specifierLoc,
   return decl;
 }
 
+void ParamDecl::setDefaultArgumentKind(DefaultArgumentKind K) {
+  assert(static_cast<DefaultArgumentKind>(Bits.ParamDecl.defaultArgumentKind) ==
+             DefaultArgumentKind::None &&
+         "Overwrite of default argument kind");
```

Comment:
I would do `getDefaultArgumentKind()` here, but it doesn't really matter

Reply:
Is an overwrite here not something we’d want to go unnoticed?

Reply:
Yes, I just mean doing `getDefaultArgumentKind()` instead of `static_cast`'ing the bits, but I don't feel too strongly about it

Reply:
Ah, I misunderstood.